x-pack/plugins/ml/server/shared_services/providers/anomaly_detectors.ts

```diff
@@ -4,40 +4,33 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ILegacyScopedClusterClient, KibanaRequest } from 'kibana/server';
+import { KibanaRequest } from 'kibana/server';
 import { Job } from '../../../common/types/anomaly_detection_jobs';
-import { SharedServicesChecks } from '../shared_services';
+import { GetGuards } from '../shared_services';
 
 export interface AnomalyDetectorsProvider {
   anomalyDetectorsProvider(
-    mlClusterClient: ILegacyScopedClusterClient,
     request: KibanaRequest
   ): {
     jobs(jobId?: string): Promise<{ count: number; jobs: Job[] }>;
   };
 }
 
-export function getAnomalyDetectorsProvider({
-  isFullLicense,
-  getHasMlCapabilities,
-}: SharedServicesChecks): AnomalyDetectorsProvider {
+export function getAnomalyDetectorsProvider(getGuards: GetGuards): AnomalyDetectorsProvider {
   return {
-    anomalyDetectorsProvider(mlClusterClient: ILegacyScopedClusterClient, request: KibanaRequest) {
-      //  APM is using this service in anomaly alert, kibana alerting doesn't provide request object
-      // So we are adding a dummy request for now
-      // TODO: Remove this once kibana alerting provides request object
-      const hasMlCapabilities =
-        request.params !== 'DummyKibanaRequest'
-          ? getHasMlCapabilities(request)
-          : (_caps: string[]) => Promise.resolve();
+    anomalyDetectorsProvider(request: KibanaRequest) {
       return {
         async jobs(jobId?: string) {
-          isFullLicense();
-          await hasMlCapabilities(['canGetJobs']);
-          return mlClusterClient.callAsInternalUser(
-            'ml.jobs',
-            jobId !== undefined ? { jobId } : {}
-          );
+          return await getGuards(request)
+            .isFullLicense()
+            .hasMlCapabilities(['canGetJobs'])
+            .ok(async ({ scopedClient }) => {
+              const { body } = await scopedClient.asInternalUser.ml.getJobs<{
```

Comment:
Is it possible that this logs errors to the Kibana log when querying for a non-existent job, which it didn't before? Is there a less noisy way of checking for a job's existence?

Reply:
yes it appears this is a change introduced with using the new js client. All errors are logged to kibana's log.
I am able to disable the log message by passing ` { ignore: [404] }` as an option to `getJobs`.
I don't know whether this is something we should be doing by default or perhaps exposing this option to consumers of our shared functions.

Reply:
For an API that returns a collection I would expect the result to just not contain the items it couldn't find. 404s/"not found" exceptions mainly make sense for single-resource APIs IMHO.

Reply:
Yes, that does make sense, unfortunately this is a basic wrapper around the elasticsearch endpoint, and so behaves in the exact same way.
https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-get-job.html